framework/src/actions/ActionFactory.C

```diff
@@ -23,6 +23,11 @@ ActionFactory::reg(const std::string & name,
                    const std::string & file,
                    int line)
 {
+  auto key = std::make_pair(name, task);
+  if (_current_objs.count(key) > 0)
+    return;
+  _current_objs.insert(key);
```

Comment:
For future reference we have a moose_try_emplace() method (C++17 wrapper) which is more efficient (and a single line) for this type of thing.